### ISSUE TITLE
Update documentation for custom actions

### ISSUE THREAD

**Issue by trych** [mzur/kirby-uniform#214]
Hi there,

I am trying to create a custom action, but the code in the docs seems to be outdated for Kirby v3's new plugin structure.
When I past [the sample code](https://kirby-uniform.readthedocs.io/en/latest/actions/actions/#custom-actions) into a plugin file (like `plugins/my-actions-plugin/index.php`), then I get the error:

```txt
Class 'Uniform\Actions\Action' not found
```

Does anybody know, what structure my plugin needs to have to create a custom action in Kirby v3?

Thanks!

**Comment by mzur** [comment]
Looks like Uniform wasn't installed correctly. Can you use `\Uniform\Form` in a controller?

**Comment by trych** [comment]
Yes, I can use all the other Uniform functionality just fine.

I think I tracked down the issue. I had this in a plugin that was alphabetically placed before the `uniform` plugin folder. When I rename my plugin `zzz` it does not give me the error anymore. It looks like Kirby is registering the plugins one by one and only after the uniform plugin is registered I can use the uniform class.

Is this expected behaviour? Or am I setting up my plugin incorrectly? According to the [Kirby v3 docs](https://getkirby.com/docs/guide/plugins/plugin-basics), plugins now should be wrapped in the static `Kirby::plugin()` method. I have not done this, as I don't know how to insert the snippet from the docs within the array that this method expects. Any ideas how to do this? I am currently not even sure if this dumpAction is working, not sure what output to expect and where.

**Comment by mzur** [comment]
I see. We had a similar issue before but with the whole plugin ( #204 ). Can you please try this code sample:

```php
<?php

use Uniform\Actions\Action;

class DumpAction extends Action
{
    public function perform()
    {
        try {
            var_dump($this->form->data());
        } catch (\Exception $e) {
            $this->fail($e->getMessage());
        }
    }
}
```

**Comment by trych** [comment]
I gave it a try, unfortunately gives me the very same error:

![20210903-121147_Screenshot_GoogleChrome](https://user-images.githubusercontent.com/9803905/131989291-d76913c3-de3f-42e3-95d7-3a5fa1c2b8dd.png)

**Comment by mzur** [comment]
Ok thanks. I assume that you didn't install Uniform using Composer? I'll have a look.

**Comment by trych** [comment]
No, I installed it as a git submodule.

**Comment by mzur** [comment]
Ok, please try this:

1. Create a file `plugins/my-actions-plugin/index.php` with the content:
```php
<?php
 
load([
    'Uniform\\Actions\\DumpAction' => 'DumpAction.php'
], __DIR__);
```
2. Create a file `plugins/my-actions-plugin/DumpAction.php` with the content:
```php
<?php

namespace Uniform\Actions;

class DumpAction extends Action
{
    public function perform()
    {
        try {
            var_dump($this->form->data());
        } catch (\Exception $e) {
            $this->fail($e->getMessage());
        }
    }
}
```

Does it work now?

**Comment by trych** [comment]
Yes, it does not give me an error anymore, and it dumps the data on submit, however for some bizarre reason I seem not to be able to change the dump action. It always keeps dumping the form data, even though when I remove (!) the line `var_dump($this->form->data());`??

I'm not sure what to make of this, is there some default built-in action behaviour that dumps the form data? Because whatever I write into the dump action it gets ignored and it just dumps the form data. This is really strange. Thought it was some cache issue, but it happens even after deleting the cache or changing browsers.

So I am *not* in fact sure, if the dumpAction really works... yes it dumps something, but I cannot change it in any way...?

**Comment by mzur** [comment]
Yes, you are right, there is an undocumented DumpAction in Uniform. Try to rename the action/class and the changes should apply.

**Comment by trych** [comment]
Oh my, I thought I was going insane. 😅

Good. Now I can report that it works. You should probably either change the undocumented Action's name or change the example in the doc's, else people will run into the same problem. Also, it might make sense to use a example in the doc that works in an Ajax setup as well, I don't think the `var_dump()` command does.

This issue is solved for me now, thanks!

**Comment by mzur** [comment]
Here is a summary of the changes required in the docs:

- [ ] Update the path to the action file (should be an `index.php` file)
- [ ] Update the `index.php` example with a custom autoloaded action class file (see above)
- [ ] Update the example action class file to something else than the dump action (should be compatible with ajax)

I'll leave this open as a reminder. Anyone willing to controbute with a PR (@trych?) is welcome to do so.

**Comment by adamkiss** [comment]
I'll look into this in the next few days.

**Comment by eXpl0it3r** [comment]
Just ran into this and thought, I was going crazy, as following the docs didn't seem to work 😄 

Do you want to differentiate between composer and non-composer setups? If yes, how do you imagine structuring it?

**Comment by adamkiss** [comment]
Huh, more than a few days than 😊 

the only additional part for nom-composer setup is explicit loading, i think, although ive been honestly putting this off for months ☹️

**Comment by eXpl0it3r** [comment]
Well yeah, my question was on how to best write the docs considering both scenarios. I guess an additional section saying "for non-composer user, also do this" or similar...

**Comment by mzur** [comment]
I'd tend to write the docs for non-Composer users only. If you want to load the custom action with Composer, you need to write it as a separate PHP package and you should already know how that is done. Or were you thinking about loading the action with a custom classmap?

**Comment by eXpl0it3r** [comment]
I didn't know how it's done for Composer users, just saw that you mentioned how it is for non-Composer users.

@adamkiss if you want, I can PR onto your branch or take your branch and create a new PR

**Comment by adamkiss** [comment]
@eXpl0it3r I've updated my PR - #217 - with non-composer setup as well, if you can review
